modules/pokemon.py

```diff
@@ -1036,12 +1025,16 @@ def level(self) -> int:
         return self.data[84]
 
     @property
-    def status_condition(self) -> StatusCondition:
+    def status_condition(self) -> tuple[StatusCondition, int]:
+        """Returns the StatusCondition of a pokémon and the number of turns remaining if known."""
         # This property is not available for boxed Pokémon
-        if len(self.data) <= 80:
-            return StatusCondition.Healthy
-        else:
-            return StatusCondition.from_bitfield(self.data[80])
+        status = StatusCondition.Healthy
+        turns = 0
+        if len(self.data) > 80:
+            status = StatusCondition.from_bitfield(self.data[80])
+            if status == StatusCondition.Sleep:
+                turns = self.data[80] & 0b0111
```

Comment:
Note: I haven't looked much into live debugging, but looking at how these bitmasks work, it's extremely likely that the turns mask applies to all status effects that last a number of turns, so I'd imagine this bitwise operation should just be done for every status effect. For now, I just kept parity with what the code was doing before.

Reply:
Thanks for the source.